src/js/actions/documents.js

```diff
@@ -328,15 +334,19 @@ define(function (require, exports) {
                     return;
                 }
 
-                var currentRef = documentLib.referenceBy.current;
-                return _getDocumentByRef(currentRef)
+                var currentRef = documentLib.referenceBy.current,
+                    documentPromise = _getDocumentByRef(currentRef),
+                    deselectPromise = descriptor.playObject(selectionLib.deselectAll());
```

Comment:
Because it is safe to do so AFAICT, and moving it up could prevent delays. If we wait and then submit that command it's worse than submitting it early to PS for parallel execution.